### PR TITLE
STORY-16952 Update API docs for whisper prompt condition

### PR DESCRIPTION
**Pull request by abhig10**
[STORY-16952 Update API docs for whisper prompt condition](https://invoca.atlassian.net/browse/STORY-16952)

Adds Invoca API documentation for setting the whisper_prompt_condition field on campaigns.

⚠️  https://github.com/Invoca/developer-docs/pull/353 must be merged and deployed before this PR.

Screenshot 2023-12-14 at 12 10 12 AM
Screenshot 2023-12-14 at 12 10 22 AM
Screenshot 2023-12-14 at 12 10 32 AM
Screenshot 2023-12-14 at 12 10 43 AM

## Checklist

- [ ] Find the [Service owning team](https://docs.google.com/spreadsheets/d/1YF2wuepY5SZTpVEdT9gwhTPhpnDGZeVjywEwvbJS5TQ/edit#gid=0&fvid=1520238175) for these changes, and tag the team as "Reviewers" on this PR
- [ ] Test the documentation changes on readthedocs as a private branch
- [ ] If changing general content, have agreement on whether to apply to latest version or all versions (if all versions, provide links to the related PRs below)

### CODE REVIEW

File: source/api_documentation/network_integration/call_recording_and_whisper_prompt_conditions.rst
Comment (omeedrj) [inline source/api_documentation/network_integration/call_recording_and_whisper_prompt_conditions.rst:1]: ⛏️ I'd suggest renaming this file name to `ivr_tree_conditions` to match the `IvrTreeCondition` class that will be in web

File: source/api_documentation/network_integration/call_recording_condition/index.rst
Comment (omeedrj) [inline source/api_documentation/network_integration/call_recording_condition/index.rst:1]: 🔧 I'm leaving this comment here, but it applies more generally for STORY-16952 as a whole, given we're supporting API versions 2019-05-01+, we're also going to need branches for `2022-03-01` and `2020-10-01`. 

(I saw the [2019-05-01 PR](https://github.com/Invoca/developer-docs/pull/353) was created 👍)

image

**Review by omeedrj — CHANGES_REQUESTED** [review]
([CREG](https://github.com/erikthedeveloper/code-review-emoji-guide))

File: source/api_documentation/network_integration/call_recording_and_whisper_prompt_conditions.rst
Comment (abhig10) [reply source/api_documentation/network_integration/call_recording_and_whisper_prompt_conditions.rst:1]: Good point. Done.

File: source/api_documentation/network_integration/call_recording_condition/index.rst
Comment (abhig10) [reply source/api_documentation/network_integration/call_recording_condition/index.rst:1]: I remembered that I added support only for `2022-08-01` in https://github.com/Invoca/developer-docs/pull/344 without realizing that `2022-03-01` or `2020-10-01` also show documentation with `2019-05-01` API version. 🤦 

Creating PRs for `2022-03-01` and `2020-10-01`.

Comment (abhig10) [reply source/api_documentation/network_integration/call_recording_condition/index.rst:1]: PR for `2020-10-01` https://github.com/Invoca/developer-docs/pull/355
PR for `2022-03-01` https://github.com/Invoca/developer-docs/pull/354

**Review by omeedrj — APPROVED** [review]
I will coordinate merging this after Ringswitch has been deployed for [STORY-16951](https://invoca.atlassian.net/browse/STORY-16951)

**Pull request merged**